### PR TITLE
Add empty technique array to rules

### LINKED ISSUE #812: [Rule Tuning] Threats which define only a tactic must contain an empty array
Related to https://github.com/elastic/kibana/issues/87546

In Kibana 7.11, ATT&CK techniques were made optional, but the field is still required to be set as an empty array. In 7.12, the schema in Kibana will be updated to make the field optional, no longer requiring an empty array. Once this is done, we can strip the empty array from the rules:

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
Resolves #812

## Summary
Inserts empty array into rule `payload` threat mapping where technique is not defined

### CODE REVIEW

File: detection_rules/rule.py [hunk 1]
```diff
@@ -316,7 +341,42 @@ def dict_hash(cls, contents, versioned=True):
...
+        rules_versions = load_versions
+
+        if self.id in rules_versions:
+            version_info = rules_versions[self.id]
+            version = version_info['version']
+            return version + 1 if self.get_hash() != version_info['sha256'] else version
+        else:
+            return 1
+
+    def get_payload(self, include_version=False, randomize_id=False, embed_metadata=False, target_version=None):
```
Comment (rw-access) [inline detection_rules/rule.py:359]: personally, I like `replace_id` or `generate_id` or something like that. randomize just seems a little odd to me for some reason

File: detection_rules/rule.py [hunk 2]
```diff
@@ -316,7 +341,42 @@ def dict_hash(cls, contents, versioned=True):
...
+        """Get rule as uploadable/API-compatible payload."""
+        from uuid import uuid4
+        from .schemas import downgrade
+
+        payload = self._run_build_time_transforms(self.contents.copy())
+
+        if include_version:
+            payload['version'] = self.get_version()
+
+        if embed_metadata:
```
Comment (rw-access) [inline detection_rules/rule.py:369]: nice, I dig this

**Review by rw-access — APPROVED** [review]
seems good

File: detection_rules/rule.py (hunk 1 above)
Comment (brokensound77) [reply detection_rules/rule.py:359]: That's right - now I remember you saying this in another PR. Will update, thanks

**Review by threat-punter — APPROVED** [review]

**Pull request merged**